### PR TITLE
Mesh non-collision loss optimization

### PR DESCRIPTION
**Pull request by qianl-nv**
## Summary
Batch mesh collision SDF queries to reduce solver time from 55s to 4s for the droid kitchen scene.

## Detailed description
- The baseline runs sphere to SDF serial per-candidate in compute_no_overlap_loss_mesh resulting in one warp launch per candidate. While the SDF kernel itself runs fast (0.75ms), the serial multi launch results in NoOverlapMESH taking 55ms per iteration (97% of each interation time).
- Replace per-candidate mesh queries with one batched Warp launch. compute_no_overlap_loss_mesh rewritten as a fully tensorized (B, P) / (B, S) pipeline (batch × pairs × spheres).
- Remove unnecessary host sync points — .item() when building yaw tensors from Python lists; if not active_pair.any(): continue forcing GPU→CPU sync every iteration.

## Tests
- B=4, mixed overlap/yaw; loss and ∇positions match baseline serial implementation
- For existing droid kitchen scene, validation pass rate for serial vs batched implementation is identical.
- Manual run droid_pick_and_place_lightwheel_kitchen.yaml and layout looks correct over 20 resets.

Command to run:
/isaac-sim/python.sh isaaclab_arena/evaluation/policy_runner.py     --viz kit     --policy_type zero_action     --num_episodes 20  --disable_fabric  --device cpu  --env_graph_spec_yaml isaaclab_arena_environments/kitchen_bench/droid_pick_and_place_lightwheel_kitchen.yaml

with solver logging enabled should see
[RelationSolver] solve: 3715.2 ms | batch=50 | objects=3 optimizable + 3 anchors | no-overlap pairs=12 | iters=600 (6.19 ms/iter)
vs baseline
[RelationSolver] solve: 56006.8 ms | batch=50 | objects=3 optimizable + 3 anchors | no-overlap pairs=12 | iters=600 (93.34 ms/iter)

### CODE REVIEW

File: isaaclab_arena/relations/mesh_pair_cache.py [hunk 1]
```diff
@@ -143,21 +143,25 @@ class MeshPairCache:
...
-        assert (
-            len(self.pair_obstacle_bbox_includes_yaw) == self.num_pairs
-        ), "pair_obstacle_bbox_includes_yaw length mismatch"
         assert len(self.pair_obstacle_is_fixed) == self.num_pairs, "pair_obstacle_is_fixed length mismatch"
         assert self.all_centers_local.shape[0] == self.total_spheres, "all_centers_local size mismatch"
         assert self.all_radii.shape[0] == self.total_spheres, "all_radii size mismatch"
         assert self.sphere_pair_id.shape[0] == self.total_spheres, "sphere_pair_id size mismatch"
         assert self.sphere_mesh_idx.shape[0] == self.total_spheres, "sphere_mesh_idx size mismatch"
         assert int(self.pair_sphere_count.sum().item()) == self.total_spheres, "pair_sphere_count sum mismatch"
+        assert self.pair_fixed_obstacle_pos_tensor.shape == (self.num_pairs, 3)
```
Comment (qianl-nv) [inline isaaclab_arena/relations/mesh_pair_cache.py]: add error message

File: isaaclab_arena/relations/no_overlap_mesh.py [hunk 2]
```diff
@@ -49,124 +49,133 @@ def compute_no_overlap_loss_mesh(
...
+        subject_rows.append([
+            ori_b.get(mesh_cache.pair_subject_objs[p], 0.0) if mesh_cache.pair_subject_applies_yaw[p] else 0.0
+            for p in range(num_pairs)
+        ])
+        obstacle_rows.append([
+            ori_b.get(mesh_cache.pair_obstacle_objs[p], mesh_cache.pair_fixed_obstacle_yaw[p]) for p in range(num_pairs)
+        ])
+    subject_yaws = torch.tensor(subject_rows, dtype=torch.float32, device=device)
+    obstacle_yaws = torch.tensor(obstacle_rows, dtype=torch.float32, device=device)
+    return subject_yaws, obstacle_yaws, True
```
Comment (qianl-nv) [inline isaaclab_arena/relations/no_overlap_mesh.py:176]: why always returning True for yaw active? is this  regression from main?

File: isaaclab_arena/relations/no_overlap_mesh.py [hunk 3]
```diff
@@ -412,24 +426,38 @@ def _finalize_mesh_cache(entries: list[MeshPairEntry], device: torch.device) ->
...
+            [e.obstacle_bbox_includes_yaw for e in entries], dtype=torch.bool, device=device
+        ),
     )
 
 
 def _rotate_bbox_extents(
     bbox_min: torch.Tensor, bbox_max: torch.Tensor, yaws: torch.Tensor
 ) -> tuple[torch.Tensor, torch.Tensor]:
-    """Return the AABB enclosing a Z-rotated bbox around the object origin."""
+    """Return the AABB enclosing a Z-rotated bbox around the object origin.
```
Comment (qianl-nv) [inline isaaclab_arena/relations/no_overlap_mesh.py:454]: revert change

File: isaaclab_arena/environments/arena_env_builder.py [hunk 4]
```diff
@@ -98,7 +98,7 @@ def _solve_relations(self) -> None:
         placer_params = self.arena_env.placer_params
         if placer_params is None:
             placer_params = ObjectPlacerParams(
-                solver_params=RelationSolverParams(verbose=False, save_position_history=False),
+                solver_params=RelationSolverParams(verbose=False, save_position_history=False, profile=True),
```
Comment (qianl-nv) [inline isaaclab_arena/environments/arena_env_builder.py]: revert profile logging change

**Review by qianl-nv — COMMENTED** [review]
self review 1

File: isaaclab_arena/environments/arena_env_builder.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (qianl-nv) [reply isaaclab_arena/environments/arena_env_builder.py]: done

File: isaaclab_arena/relations/no_overlap_mesh.py (hunk 2 above)
Comment (qianl-nv) [reply isaaclab_arena/relations/no_overlap_mesh.py:176]: early return might be False. keep

**Review by zhx06 — APPROVED** [review]
LGTM! Thanks for making the performance much better!
